pkg/algorithm/nodeselect/v1alpha1/select_node.go

```diff
@@ -198,18 +239,40 @@ func (ac *Config) poolType() string {
 
 // getBlockDevice return the all disks of a certain type(e.g. sparse, blockdevice) which is specified in spc.
 func (ac *Config) getBlockDevice() (*ndmapis.BlockDeviceList, error) {
-	diskFilterLabel := diskFilterConstraint(ac.Spc.Spec.Type)
-	bdL, err := ac.BlockDeviceClient.List(metav1.ListOptions{LabelSelector: diskFilterLabel})
+	var bdl *blockdevice.BlockDeviceList
+	diskType := ac.Spc.Spec.Type
+	diskFilterLabel := diskFilterConstraint(diskType)
+	bdList, err := ac.BlockDeviceClient.List(metav1.ListOptions{LabelSelector: diskFilterLabel})
 	if err != nil {
 		return nil, err
 	}
-	bdl := bdL.Filter(blockdevice.FilterNonInactive)
+
+	if ProvisioningType(ac.Spc) == ProvisioningTypeManual {
+		bdl = bdList.Filter(blockdevice.FilterNonInactive)
+	} else {
+		bdl, err = ac.getBlockDeviceListInAutoMode(bdList, diskType)
+		if err != nil {
+			return nil, err
+		}
+	}
 	return bdl.BlockDeviceList, nil
 }
 
-//TODO: Make changes in below code befor PR checked in
+func (ac *Config) getBlockDeviceListInAutoMode(bdList *blockdevice.BlockDeviceList, diskType string) (*blockdevice.BlockDeviceList, error) {
+	var bdl *blockdevice.BlockDeviceList
+	if diskType == string(apis.TypeSparseCPV) {
+		bdl = bdList.Filter(blockdevice.FilterNonInactive, blockdevice.FilterNonPartitions, blockdevice.FilterSparseDevices)
+	} else {
+		bdl = bdList.Filter(blockdevice.FilterNonInactive, blockdevice.FilterNonPartitions, blockdevice.FilterNonSparseDevices)
```

Comment:
Instead of filtering for NonPartitions, we should be filtering for NonFSType

Reply:
Updated